### PR TITLE
Fix redundant declarations warnings

### PR DESCRIPTION
**Pull request by AdrianDC**

### CODE REVIEW

File: include/wx/utils.h [hunk 1]
```diff
@@ -601,8 +601,8 @@ WXDLLIMPEXP_BASE bool wxGetDiskSpace(const wxString& path,
                                      wxDiskspaceSize_t *pTotal = NULL,
                                      wxDiskspaceSize_t *pFree = NULL);
 
-
-
+// wxQsort can be initially declared in wx/vector.h
```
Comment (paulcor) [inline include/wx/utils.h]: Only if wxUSE_STD_CONTAINERS == 0. Would probably be better to use a specific preprocessor symbol to indicate whether wxQsort is declared, something like WXQSORT_DECLARED perhaps.

**Comment by vadz** [comment]
Sorry, I don't understand these changes at all. What problem(s) are they supposed to correct? And where do all these symbols being checked come from (especially `__NO_ISOCEXT` which I don't think I've ever seen before)?

File: include/wx/utils.h (hunk 1 above)
Comment (AdrianDC) [inline include/wx/utils.h]: Done

**Comment by AdrianDC** [comment]
I added more details about the constants' origins in the commits details and the origin C headers of their inheritance.

**Comment by vadz** [comment]
Oh, I see, all this is just to avoid `-Wredundant-decls` warnings?

Sorry, but if this is the case, I just don't think it's worth the extra ugliness. Moreover, things like `_STDIO_H_` are completely non-standard, of course, so this doesn't even fix the warnings on all systems -- just on the one you tested with (and even not for all such systems, but for those using this particular flavour of CRT). If we really wanted to do this cleanly, we'd need to define our own symbols like `wxSTDIO_H_INCLUDED` etc i.e. it would require even more changes.

IMO you should just disable `-Wredundant-decls` before including wx headers and reenable it afterwards if you really care about this (IME not useful) warning. At most, I'd be ready to add some `wxGCC_WARNING_{SUPPRESS,RESTORE}()` to our own headers. But I disagree with working around this warning in this way.

**Pull request closed without merging**